src/Controls/src/SourceGen/SetPropertyHelpers.cs

```diff
@@ -58,6 +58,27 @@ public static void SetPropertyValue(IndentedTextWriter writer, ILocalValue paren
 			return;
 		}
 
+		// Heuristic: If BP is null but the type has a property/field with a BindablePropertyAttribute,
+		// assume the BP will be generated by another source generator
+		// This is checked after BP-based mechanisms but before property setter to ensure
+		// that properties with source-generated BPs use SetBinding/SetValue instead of the setter
+		if (!asCollectionItem && bpFieldSymbol == null && 
+			parentVar.Type.HasBindablePropertyHeuristic(localName, context, out var explicitPropertyName))
```

Comment:
this test should be within CanSetValue and CanSetBinding, leaving the structure of this method unchanged